code-challenges.test.js

```diff
@@ -15,43 +17,115 @@
+////xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
 // a) Create a test with expect statements for each of the variables provided.
-
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
+// var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67 8, 24]
+// var nums2 = [109, 5, 9, 67 8, 24]
 // Expected output: [5, 109]
+// b) Create the function that makes the test pass.
 
 
-
-// b) Create the function that makes the test pass.
+// // creat a test, let test fail.
+//     // test will test will take in an array and expect a SORTED array
+// //fn will take in array and sort them from min to MAX .sort a-b
+// // fn will ONLY ONLY ONLY ONLY>>>>> return 2 numbers min and MAX
+// // might need an empty array.
+
+describe("an Array of numbers", () => {
+    it("only the smallest number and largest number", () => {
+      expect(numsArray([3, 56, 90, -8, 0, 23, 6])).toEqual([-8, 90])
+      expect(numsArray([109, 5, 9, 67, 8, 24])).toEqual([5, 109])
+    })
+})
+const numsArray = (array) =>{
+    sortedNums = array.sort((a, b) => a-b)
+    lilNum = sortedNums.shift()// so i managed to isloted each value that I wanted, .concat is not working
+    bigNum = sortedNums.pop()
+    answerArray =[lilNum,bigNum] //after a very long while, I thought will maybe its as easy as calling my new var's into an array and yes, yes it is. 
+    // console.log(bigNum)
+    // console.log(lilNum)
+    return answerArray
+}
+var nums1 = [3, 56, 90, -8, 0, 23, 6]
+console.log(numsArray(nums1))
+// // Expected output: [-8, 90]
+var nums2 = [109, 5, 9, 67, 8, 24]
+console.log(numsArray(nums2))
+// // Expected output: [5, 109]
 
 
+////xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
-
 // a) Create a test with an expect statement using the variables provided.
 
+//create a test and let it fail
+//function
+    // will take in testArray1,2
+    // concat two arrays then sort?
+    // we sort then concat
+//duplicates   
+    //maybe a for loop simulir to febanchi 
+    //.filiter
+    // going to need any empty array to hold new numbers
+
+
+describe("function that test two arrays", () => {
+    test("takes in two arrays of numbers, returns ONE array with no duplicates", () => {
+      expect(twins([3, 7, 10, 5, 4, 3, 3],[7, 8, 2, 3, 1, 5, 4])).toContain(3, 7, 10, 5, 4, 8, 2, 1)
+    })
+})
+const twins = (array1, array2)=>{
+    newArray = (array2.concat(array1)).sort((a,b) => a-b)
+    noDubs = []
+    for(let i=0; i<newArray.length; i++){
+        if(newArray[i] !== newArray[i+1])// I really don't know how I did it but I did
```

Comment:
Awesome! I would love for you to teach this to someone-- it'll really stick in your head then.